MIP/mip-1/README.md

```diff
@@ -0,0 +1,113 @@
+
+### Practical Considerations
+In practice, there are several phenomena regarded as external to the ENTL protocol which need to be considered in order to make it safely usable. 
+
+#### Securing the Nonce
+In order to avoid the possibility of malicious usage, both the current nonce and the next nonce need to be stored in a manner that is difficult to access except by the current program. We do not intend to impose cryptographically secure requirements on this storage, as that begins to invoke concepts related to the literature on succinct arguments for programs. However, there are several simpler ways that can make accessing and using the nonce difficult.
+
+To describe these, it first helps to consider what enables a nonce attack and to consider the time these operations take. 
+
+1. The attacker must first obtain the nonce which takes some duration $d_E = t_E' - t_E$. 
+2. The attacker then must use the nonce to send in an APP message which the Enclave will process. This takes some duration $d_S = t_S' - t_S$.
+3. In total, we may say the attack takes some duration $d_A = t_S' - t_E'$. 
+4. Meanwhile, the legitimate Client will have some duration $d_L = t_L' - t_L$ over which the nonce is stored as either the current or next nonce and has not yet been used. 
+5. The period between when the legitimate Client first stores this nonce and when the Enclave would process it is bounded by a time $t_S'$ wherein the attacker sends the malicious APP message and $t_L$. 
+
+This model of an attack leads to several prevention strategies:
+
+- **High Temporal Resolution**: frequently updating the nonce narrows the window $d_L$ in which the attacker can obtain the nonce and use it before the legitimate Client does. If this exceeds, the time to obtain the nonce and send it, i.e., $d_A$, the attack is impossible. It may thus be ideal for the Client to update the nonce even without a message that corresponds to is APP logic. 
+- **Security via Obscurity**: the implication throughout this MIP is that Client stores its nonce in program memory. As program memory is highly-dynamic, particularly on the stack, it would be difficult to identify the location of the nonce in memory, increasing $d_A$. This is a form of security via obscurity which could be further upgraded by data structures which further randomize the location of the nonce in memory.
```

Comment:
I don't think this is a valid strategy. If an attacker is able to access the Client process memory, the assumption should be that they can easily discover the location in which the nonce is stored by following the data structures in the program's state.